Analysis/src/AnaTupleReader.cpp

```diff
@@ -258,12 +264,29 @@ void AnaTupleReader::DefineBranches(const NameSet& active_var_names, bool all, c
     DefineP4(df, "VBF1");
     DefineP4(df, "VBF2");
 
+
+
+
     const auto convertToDataId = [&](unsigned dataset, unsigned event_region, int unc_source, int unc_scale) {
         return DataId(GetDatasetByHash(dataset), GetRegionByHash(event_region),
                       static_cast<UncertaintySource>(unc_source), static_cast<UncertaintyScale>(unc_scale));
     };
     Define(df, "dataId", convertToDataId, { "dataset", "event_region", "unc_source", "unc_scale" }, true);
 
+    const auto Full_branch_name=[&](std::string branch_name){
```

Comment:
- this functionality should also be used in ProcessAnaTuple -> define it as a method instead of lambda
- if full_name is not found - throw an exception
- the range for i not consistent between the method and how you define nicknames above.

Reply:
This is the only change I didn't do.. because: if it's a method outside of AnatupleReader, it can't read the df (defined inside AnaTupleReader) anymore, should I define outside and re-open the files? I'm not so sure about it.. And I don't understand why it must be used also in ProcessAnaTuple